src/hydra-eval-jobs/hydra-eval-jobs.cc

```diff
@@ -440,11 +440,21 @@ int main(int argc, char * * argv)
             auto named = job.find("namedConstituents");
             if (named == job.end()) continue;
 
+            std::unordered_map<std::string, std::string> brokenJobs;
             if (myArgs.dryRun) {
                 for (std::string jobName2 : *named) {
                     auto job2 = state->jobs.find(jobName2);
-                    if (job2 == state->jobs.end())
-                        throw Error("aggregate job '%s' references non-existent job '%s'", jobName, jobName2);
+                    if (job2 == state->jobs.end()) {
+                        printError("aggregate job '%s' references non-existent job '%s'", jobName, jobName2);
+                        brokenJobs[jobName2] = "does not exist";
+                        continue;
+                    }
+                    if (job2->find("error") != job2->end()) {
+                        std::string error = (*job2)["error"];
+                        printError("aggregate job '%s' references broken job '%s': %s", jobName, jobName2, error);
+                        brokenJobs[jobName2] = error;
+                        continue;
+                    }
```

Comment:
Would it be possible to make this its own function? I'm not sure it wouldbe straight forward to do so, but if it were ...